core/src/com/ericc/the/game/systems/realtime/RenderSystem.java

```diff
@@ -10,36 +10,42 @@
 import com.badlogic.gdx.graphics.g2d.SpriteBatch;
 import com.badlogic.gdx.graphics.g2d.TextureRegion;
 import com.badlogic.gdx.math.Affine2;
-import com.badlogic.gdx.math.Vector2;
 import com.badlogic.gdx.utils.viewport.Viewport;
 import com.ericc.the.game.Mappers;
 import com.ericc.the.game.Media;
 import com.ericc.the.game.TileTextureIndicator;
+import com.ericc.the.game.components.FieldOfViewComponent;
 import com.ericc.the.game.components.PositionComponent;
+import com.ericc.the.game.components.ScreenBoundariesComponent;
 import com.ericc.the.game.components.SpriteSheetComponent;
+import com.ericc.the.game.entities.Player;
+import com.ericc.the.game.entities.Screen;
 import com.ericc.the.game.map.Map;
+import com.ericc.the.game.shaders.GrayscaleShader;
 
 import java.util.ArrayList;
 
-import static java.lang.Integer.max;
-import static java.lang.Integer.min;
-
 /**
  * The system responsible for drawing the map and the entities on the screen.
  * It is desirable to concentrate and contain graphics-related code here.
  */
 public class RenderSystem extends EntitySystem {
     private Map map;
     private Viewport viewport;
+    private ScreenBoundariesComponent visibleMapArea;
     private final Affine2 transform = new Affine2();
 
     private SpriteBatch batch = new SpriteBatch();
+    private SpriteBatch tilesSeen = new SpriteBatch();
     private ImmutableArray<Entity> entities; // Renderable entities.
+    private FieldOfViewComponent playersFieldOfView;
 
-    public RenderSystem(Map map, Viewport viewport) {
+    public RenderSystem(Map map, Viewport viewport, Player player, Screen screen) {
         super(9999); // Rendering should be the last system in effect.
         this.map = map;
         this.viewport = viewport;
+        this.playersFieldOfView = Mappers.fov.get(player);
```

Comment:
Może niech to od razu bierze playersFieldOfView skoro argument player nie jest do niczego innego używany?